src/system.ts

```diff
@@ -308,23 +299,23 @@ export async function updateAndSaveCurrentReport() {
     reportCache.set('current', report);
     log.info('Report cached');
 
+    log.info('Getting observers from contract state...');
     // Get selected observers for the current epoch from the contract
-    let observers: string[] = [];
-    try {
-      log.info('Getting observers from contract state...');
-      observers = (await prescribedObserversSource?.getObservers()) ?? [];
-      log.info(`Retreived ${observers.length} observers from contract state`);
-      if (observers.length === 0) {
-        log.error('No observers found in contract state');
-        return;
-      }
-    } catch (error: any) {
-      log.error('Unable to get observers from contract state:', {
-        message: error.message,
-        stack: error.stack,
+    const observers: string[] = await networkContract
+      .getPrescribedObservers()
+      .then((observers: WeightedObserver[]) => {
+        log.info(`Retrieved ${observers.length} observers from contract state`);
+        return observers.map(
+          (observer: WeightedObserver) => observer.observerAddress,
+        );
+      })
+      .catch((error: any) => {
+        log.error('Unable to get observers from contract state:', {
+          message: error.message,
+          stack: error.stack,
+        });
+        return [];
```

Comment:
i could add something like below to avoid computing entropy if observers is empty
```

    if (observers.length === 0) {
      log.warn(
        'No observers observers retrieved from the contract, not saving report.',
      );
      return;
    }
```

Reply:
[4094381](https://github.com/ar-io/ar-io-observer/pull/34/commits/409438137a47fa6b63ccd1e1784f5d295741075d)